File: Arithmetic/P32.cpp
#include <iostream>

int gcd(int, int);

int main(){
    std::cout << "Enter numbers separated by space : ";

    int a, b;
    std::cin >> a >>b;

    std::cout << "The GCD of " << a << " and " << b << " is : " << gcd(a,b);
    return 0;
}

int gcd (int a, int b){
    return  (b == 0) ? a : gcd(b, a % b);
}

File: Python/P32.py
def gcd(a, b) :
    '''
    Gives GCD of the given two number.

    Uses recursion to find the GCD of two numbers.

    Parameters :
        a (int) : First Number.
        b (int) : Second Number.

    Returns :
        Integer which is GCD of given two numbers.
    '''
    
    if(b==0) : return  a #Base condition
    else : return gcd(b, a%b)

if __name__ == '__main__' :
    a = int(input("Enter first number : "))
    b = int(input("Enter Second number : "))

    print("The GCD of {} and {} is : {}".format(a, b, gcd(a, b)))